### ISSUE TITLE
Add definition of substructure

### ISSUE THREAD

**Issue by vaibhavkarve** [vaibhavkarve/igl2020#7]
Noble suggested a definition for M1 "is a substructure of" M2, provided they are structures on the same language L.

**Comment by vaibhavkarve** [comment]
@noblew You were asking me if there is a test for "type A is a subtype of type B". I believe one way to implement this is as follows:
```lean
/--Check if A is a subtype of B.-/
def is_subtype_of (A B : Type) : Prop :=
  ∃ p : B → Prop, A = subtype p
```
And here is an example that uses this property to prove that `fin n` is a subtype of ℕ.
```lean
example (n : ℕ) : is_subtype_of (fin n) ℕ :=
begin
  fconstructor,
  intros m,
  exact m < n,
  refl,
end
```

**Comment by vaibhavkarve** [comment]
Closed by #72
